hail/python/hail/methods/qc.py

```diff
@@ -442,10 +442,16 @@ def get_idx(struct):
     def concordance_array(counter):
         return hl.range(0, 5).map(lambda i: hl.range(0, 5).map(lambda j: counter.get(i + 5 * j, 0)))
 
+    discordant_indices = set()
+    for i in range(5):
+        for j in range(5):
+            if i > 1 and j > 1 and i != j:
+                discordant_indices.add(i + 5 * j)
+
     def n_discordant(counter):
         return hl.sum(
             hl.array(counter)
-                .filter(lambda tup: ~hl.literal({i ** 2 for i in range(5)}).contains(tup[0]))
+                .filter(lambda tup: hl.literal(discordant_indices).contains(tup[0]))
```

Comment:
Just to make sure I understand, the issue here was that the old method didn't handle 0 and 1 correctly? Hence the need to do the `if i > 1 and j > 1` big above?

Reply:
no, it was just totally wrong!

Reply:
but I think we intended to make n_discordant exclude missingness/no data